src/sentry/static/sentry/app/components/dropdownAutoComplete.jsx

```diff
@@ -0,0 +1,163 @@
+  static propTypes = {
+    items: PropTypes.array,
+    isOpen: PropTypes.bool,
+    onSelect: PropTypes.func,
+    children: PropTypes.func,
+  };
+
+  static defaultProps = {
+    isOpen: false,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: this.props.isOpen,
+    };
+  }
+
+  toggleMenu = () => this.setState({isOpen: !this.state.isOpen});
+
+  openMenu = e => this.setState({isOpen: true});
+
+  onSelect = selectedItem => {
+    this.setState({selectedItem});
+    if (this.props.onSelect) this.props.onSelect(selectedItem);
+    this.toggleMenu();
+  };
+
+  ungroupItems = itemsToUngroup =>
+    itemsToUngroup.reduce((accumulator, {group, items}) => {
+      return [
+        ...accumulator,
+        {...group, group: true},
+        ...items.map(g => ({...g, groupedWith: group.value})),
+      ];
+    }, []);
+
+  filterItems = (items, inputValue) =>
+    items.filter(
+      i => i.group || i.value.toLowerCase().indexOf(inputValue.toLowerCase()) > -1
+    );
+
+  removeUnusedGroups = items =>
+    items.filter(i => !i.group || (i.group && items.some(g => i.value == g.groupedWith)));
+
+  autocompleteFilter = (items, inputValue) => {
+    const transformedItems = items[0].items
+      ? this.removeUnusedGroups(this.filterItems(this.ungroupItems(items), inputValue))
+      : this.filterItems(items, inputValue);
+
+    let itemCounter = 0;
+    return transformedItems.map(i => {
+      return !i.group ? {...i, index: itemCounter++} : i;
+    });
+  };
+
+  render() {
+    return (
+      <div style={{position: 'relative', display: 'inline-block'}}>
+        {this.state.isOpen && (
+          <StyledMenu>
+            <AutoComplete itemToString={item => item.searchKey} onSelect={this.onSelect}>
+              {({
+                getRootProps,
+                getInputProps,
+                getMenuProps,
+                getItemProps,
+                inputValue,
+                selectedItem,
+                highlightedIndex,
+                isOpen,
+              }) => {
+                return (
+                  <div {...getRootProps()}>
+                    <StyledInputContainer>
+                      <StyledInput
+                        autoFocus
+                        {...getInputProps({onBlur: this.toggleMenu})}
+                      />
+                    </StyledInputContainer>
+                    <div {...getMenuProps()}>
+                      <div>
+                        {this.autocompleteFilter(this.props.items, inputValue).map(
+                          (item, index) =>
+                            item.group ? (
+                              <StyledLabel key={index}>{item.label}</StyledLabel>
+                            ) : (
+                              <StyledItem
+                                key={index}
+                                highlightedIndex={highlightedIndex}
+                                index={item.index}
```

Comment:
maybe item.index and index should have more different names? such as highlightingIndex and index?